enhancements/sig-architecture/58-addon-placement-annotation/README.md

```diff
@@ -0,0 +1,295 @@
+to refactor the `ConfigCoordinates` to
+
+``` golang
+// ConfigCoordinates contains enough information to let one addon identify its configuration object.
+// A config object must be in the same namespace with its addon, or be cluster-scoped, so there is no
+// namespace field.
+type ConfigCoordinates struct {
+	// API version of the referent.
+	APIVersion string `json:"apiVersion"`
+	// Kind of the referent.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+	Kind string `json:"kind"`
+	// Name of the referent.
+	// More info: http://kubernetes.io/docs/user-guide/identifiers#names
+	Name string `json:"name"`
+}
+```
+
+with this change, the `ConfigCoordinates` will be able to support all types of configuration, and the developers will
+be able to get the `GroupVersionResource` of the configuration object.
+
+In the `ManagedClusterAddOn` API, we will also introduce `AddOnConfiguration` option, When addon has its own
+configuration, the addon can use this option to locate its configuration
+
+```golang
+// ManagedClusterAddOnSpec defines the install configuration of
+// an addon agent on managed cluster.
+type ManagedClusterAddOnSpec struct {
+	// installNamespace is the namespace on the managed cluster to install the addon agent.
+	// If it is not set, open-cluster-management-agent-addon namespace is used to install the addon agent.
+	// +optional
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+	InstallNamespace string `json:"installNamespace,omitempty"`
+
+	// addOnConfiguration is a configuration object reference that is used by current addon.
+	// +optional
+	AddOnConfiguration ConfigCoordinates `json:"addOnConfiguration,omitempty"`
+}
+```
+
+As a specific case, sometimes user want to assign their addons on specific nodes, we want to introduce a `AddOnDeployementConfig`
+to support this.
+
+```golang
+// AddOnDeployementConfig represents a deployment configuration for an add-on.
+// Currently, user can use this to configure nodeSelector, toleration and variables.
+type AddOnDeployementConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec represents a desired configuration for an add-on.
+	// +required
+	Spec AddOnDeployementConfigSpec `json:"spec"`
+}
+
+type AddOnDeployementConfigSpec struct {
+	// CustomizedVariables is a list of name-value variables for the current add-on.
+	// add-on can use these variables to render its deployment.
+	// The default is an empty list.
+	// +optional
+	CustomizedVariables []CustomizedVariable `json:"customizedVariable,omitempty"`
+
+	// NodeSelector defines which Nodes the Pods are scheduled on.
+	// The default is an empty list.
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
```

Comment:
I think we should put it in a structure working as a duck type maybe. Some thing like nodeSchedulable

Reply:
separate to a struct